pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1051,7 +1064,7 @@ func (self *awsInstance) getMountDevice(volumeID string, assign bool) (assigned
 
 	if chosen == "" {
 		glog.Warningf("Could not assign a mount device (all in use?).  mappings=%v, valid=%v", deviceMappings, valid)
-		return "", false, nil
+		return "", false, fmt.Errorf("Too many EBS volumes attached to node %s.", instanceName)
```

Comment:
Ah - OK.   I had meant for this not to be an error, but for the caller to handle.  But your way is better (and I forgot to have the caller handle, so my way plainly is too hard!)